overlay/tun_linux.go

```diff
@@ -488,7 +490,13 @@ func (t *tun) watchRoutes() {
 	rch := make(chan netlink.RouteUpdate)
 	doneChan := make(chan struct{})
 
-	if err := netlink.RouteSubscribe(rch, doneChan); err != nil {
+	netlinkOptions := netlink.RouteSubscribeOptions{
+		ReceiveBufferSize: t.useSystemRoutesBuffer,
+		ReceiveBufferForceSize: true,
```

Comment:
```suggestion
		ReceiveBufferForceSize: t.useSystemRoutesBuffer != 0,
```

I poked through the function call stack that `netlink.RouteSubscribe` makes and compared it to `netlink.RouteSubscribeOptions` and this _should_ be fine. The only difference is this always passes `true` for `ReceiveBufferForceSize`, where the default is `false`, but `ReceiveBufferForceSize` is only considered if `ReceiveBufferSize !=0`.

If we want to be really really safe and consistent with the `netlink` calls, we could take this change, but I'm not convinced it's really needed.

Reply:
fixed this just to pass the same parameters to netlink, but netlink use ReceiveBufferForceSize only when ReceiveBufferSize !=0 too.